pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -383,19 +421,21 @@ func (c *Controller) handleObject(obj interface{}) {
 	}
 }
 
-// newDeployment creates a new Deployment for a EtcdStorage resource. It also sets
+// newReplicaSet creates a new Deployment for a EtcdStorage resource. It also sets
 // the appropriate OwnerReferences on the resource so handleObject can discover
 // the EtcdStorage resource that 'owns' it.
-func newDeployment(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.Deployment {
+func (c *Controller) newReplicaSet(etcdstorage *samplev1alpha1.EtcdStorage) *appsv1.ReplicaSet {
```

Comment:
optional recommendation: I would tighten this function to a private helper that takes explicit args for its points of variability.  That way any change will force compile-time errors through your code instead of silent and weird failures.

If you want to do it, followup is fine.

Reply:
I like this approach. Created issue #11 to track this.